src/samplers.jl

```diff
@@ -0,0 +1,151 @@
+
+
+function sampleSMC!(pc::ParticleContainer,resampler::Function =resample_systematic ,resampler_threshold::AbstractFloat = 0.5)
```

Comment:
I'm not fond of having functions defined with the sampler name in the signature. Turing and most of the packages under it's umbrella use multiple dispatch on some kind of sampler object -- I think good practice would be to do the same here so that everything is consistent across the Turingverse. 

Check out [`AbstractMCMC`](https://github.com/TuringLang/AbstractMCMC.jl/blob/master/src/AbstractMCMC.jl) and the interface methods it provides for a sampler. There's also a fairly detailed writeup about the interface methodology [here](https://github.com/TuringLang/Turing.jl/blob/csp/interface-guide/docs/src/using-turing/interface.md) and a slightly less detailed one [here](https://github.com/TuringLang/Turing.jl/issues/895#issuecomment-530942507).

In this case, I think `pc`, `resampler`, and `resampler_threshold` could be bundled into one `SMCSampler` struct or something.

Reply:
I see, thanks! Is this not missleading because there is already a SMC sampler defined in Turing. Additionally, these sampler to not have the same structure as the Turing Sampler.